util.go

```diff
@@ -86,3 +87,26 @@ func appendComment(b []byte, name string) []byte {
 	name = strings.ReplaceAll(name, `*/`, `*\/`)
 	return append(b, fmt.Sprintf("/* %s */ ", name)...)
 }
+
+// queryCommentCtxKey is a context key for setting a query comment on a context instead of calling the Comment("...") API directly
+type queryCommentCtxKey struct{}
+
+// ContextWithComment returns a context that includes a comment that may be included in a query for debugging
+//
+// If a context with an attached query is used, a comment set by the Comment("...") API will be overwritten.
+func ContextWithComment(ctx context.Context, comment string) context.Context {
```

Comment:
Would you mind renaming it to WithComment to keep it consistent with WithCancel / WithTimeout and similar naming conventions?

Reply:
Done!